### PR TITLE
Extracted NestedLoopJoin from Join and renamed it to JoinPlanBuilder.

### PR DESCRIPTION
**Pull request by andreidan**

### CODE REVIEW

File: sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java [hunk 1]
```diff
@@ -359,15 +359,15 @@ private String printPlan(LogicalPlan plan) {
                 sb.append("]\n");
                 plan = aggregate.source;
             }
-            if (plan instanceof Join) {
-                Join join = (Join) plan;
+            if (plan instanceof NestedLoopJoin) {
+                NestedLoopJoin nestedLoopJoin = (NestedLoopJoin) plan;
                 startLine("Join[\n");
```
Comment (matriv) [inline sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java]: NestedLoopJoin

File: sql/src/main/java/io/crate/planner/operators/NestedLoopJoin.java [hunk 2]
```diff
@@ -40,45 +35,33 @@
...
-import java.util.Objects;
 import java.util.Set;
 import java.util.function.Consumer;
-import java.util.stream.Stream;
 
 import static io.crate.planner.operators.Limit.limitAndOffset;
 import static io.crate.planner.operators.LogicalPlanner.NO_LIMIT;
 
-public class Join extends TwoInputPlan {
+public class NestedLoopJoin extends TwoInputPlan {
```
Comment (matriv) [inline sql/src/main/java/io/crate/planner/operators/NestedLoopJoin.java]: could be package private?

File: sql/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+public class JoinPlanBuilder implements LogicalPlan.Builder {
```
Comment (matriv) [inline sql/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java]: Please add some documentation why this is extracted (different plans NestedLoop, HashJoin, etc...)

Comment (mfussenegger) [reply sql/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java]: I think it's fairly clear what this does for now. Maybe we can later add something once there is another algorithm in use and explain how the decision is made.

**Comment by andreidan** [comment]
@mfussenegger I think we can do it against master (and the sooner it gets there, the better)

**Comment by matriv** [comment]
Yep agreed, we can do it in master.

**Pull request merged**